core/store/models/ocrkey/key_bundle.go

```diff
@@ -0,0 +1,252 @@
+}
+
+// PublicKeyAddressOnChain returns public component of the keypair used in
+// SignOnChain
+func (pk *KeyBundle) PublicKeyAddressOnChain() OnChainSigningAddress {
+	return pk.onChainSigning.Address()
+}
+
+// PublicKeyOffChain returns the pbulic component of the keypair used in SignOffChain
+func (pk *KeyBundle) PublicKeyOffChain() OffChainPublicKey {
+	return OffChainPublicKey(pk.offChainSigning.PublicKey())
+}
+
+// PublicKeyConfig returns the public component of the keypair used in ConfigKeyShare
+func (pk *KeyBundle) PublicKeyConfig() [curve25519.PointSize]byte {
+	rv, err := curve25519.X25519(pk.offChainEncryption[:], curve25519.Basepoint)
+	if err != nil {
+		log.Println("failure while computing public key: " + err.Error())
+	}
+	var rvFixed [curve25519.PointSize]byte
+	copy(rvFixed[:], rv)
+	return rvFixed
+}
+
+// Encrypt combines the KeyBundle into a single json-serialized
+// bytes array and then encrypts
+func (pk *KeyBundle) Encrypt(auth string) (*EncryptedKeyBundle, error) {
+	return pk.encrypt(auth, defaultScryptParams)
+}
+
+// encrypt combines the KeyBundle into a single json-serialized
+// bytes array and then encrypts, using the provided scrypt params
+// separated into a different function so that scryptParams can be
+// weakened in tests
+func (pk *KeyBundle) encrypt(auth string, scryptParams scryptParams) (*EncryptedKeyBundle, error) {
+	marshalledPrivK, err := json.Marshal(&pk)
+	if err != nil {
+		return nil, err
+	}
+	cryptoJSON, err := keystore.EncryptDataV3(
+		marshalledPrivK,
+		[]byte(adulteratedPassword(auth)),
+		scryptParams.N,
+		scryptParams.P,
+	)
+	if err != nil {
+		return nil, errors.Wrapf(err, "could not encrypt ocr key")
+	}
+	encryptedPrivKeys, err := json.Marshal(&cryptoJSON)
+	if err != nil {
+		return nil, errors.Wrapf(err, "could not encode cryptoJSON")
+	}
+	return &EncryptedKeyBundle{
+		ID:                    pk.ID,
+		OnChainSigningAddress: pk.onChainSigning.Address(),
+		OffChainPublicKey:     pk.offChainSigning.PublicKey(),
+		EncryptedPrivateKeys:  encryptedPrivKeys,
+	}, nil
+}
+
+// Decrypt returns the PrivateKeys in e, decrypted via auth, or an error
+func (encKey *EncryptedKeyBundle) Decrypt(auth string) (*KeyBundle, error) {
+	var cryptoJSON keystore.CryptoJSON
+	err := json.Unmarshal(encKey.EncryptedPrivateKeys, &cryptoJSON)
+	if err != nil {
+		return nil, errors.Wrapf(err, "invalid cryptoJSON for OCR key bundle")
+	}
+	marshalledPrivK, err := keystore.DecryptDataV3(cryptoJSON, adulteratedPassword(auth))
+	if err != nil {
+		return nil, errors.Wrapf(err, "could not decrypt OCR key bundle")
+	}
+	var pk KeyBundle
+	err = json.Unmarshal(marshalledPrivK, &pk)
+	if err != nil {
+		return nil, errors.Wrapf(err, "could not unmarshal OCR key bundle")
+	}
+	pk.ID = encKey.ID
+	return &pk, nil
+}
+
+// MarshalJSON marshals the private keys into json
+func (pk *KeyBundle) MarshalJSON() ([]byte, error) {
+	rawKeyData := keyBundleRawData{
+		EcdsaX:             *pk.onChainSigning.X,
```

Comment:
`EcdsaX/Y` could be rehydrated from `EcdsaD` alone, using `curve.ScalarBaseMult`. It's only a 64-byte savings, though, so maybe not worth it.

Reply:
Great call - I should have done that in the first place. Going to make that change.